drivers/console/uart_mux.c

```diff
@@ -185,6 +182,19 @@ static void uart_mux_rx_work(struct k_work *work)
 	if (ret < 0) {
 		LOG_DBG("Cannot flush ring buffer (%d)", ret);
 	}
+
+	return -EAGAIN;
+}
+
+static void uart_mux_rx_work(struct k_work *work)
+{
+	struct uart_mux *uart_mux =
+		CONTAINER_OF(work, struct uart_mux, rx_work);;
+	int ret = -EAGAIN;
+
+	while (ret == -EAGAIN) {
```

Comment:
here you could have
```
do {
....
} while (ret == -EAGAIN);
```
and then you do not need to set initial value to ret variable.

Reply:
Changed as requested.